x/bank/src/keeper/mod.rs

```diff
@@ -309,19 +118,14 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module>
             }
         }
 
-        // TODO: does the SDK sort these?
-        for coin in total_supply {
-            self.set_supply(
-                ctx,
-                UnsignedCoin {
-                    denom: coin.0,
-                    amount: coin.1,
-                },
-            )
-            .unwrap_gas();
+        // does the SDK sort these?
+        // No. It uses ordering from balances https://github.com/cosmos/cosmos-sdk/blob/d3f09c222243bb3da3464969f0366330dcb977a8/x/bank/keeper/genesis.go#L32-L34
```

Comment:
We use a HashMap so we can't guarantee the same order

Reply:
Okay. Vec is a bad idea.

Reply:
Will create a new PR with dedicated structure for maybe empty coins vec